stage5.2_fuller_application/pycasa/ui/image_folder_editor.py

```diff
@@ -34,3 +35,9 @@ def _get_name(self):
 
     def _get_tooltip(self):
         return self.obj.path
+
+    @observe("closing")
```

Comment:
This is necessary. If the editor is closed while a computation is active (executing or cancelling), we get the following traceback:
```
Traceback (most recent call last):
  File "/Users/swahal/.edm/envs/ets_tutorial/lib/python3.6/site-packages/traits/trait_notifiers.py", line 342, in __call__
    self.handler(*args)
  File "/Users/swahal/.edm/envs/ets_tutorial/lib/python3.6/site-packages/traitsui/qt4/editor.py", line 86, in _e
nabled_changed
    self._enabled_changed_helper(self.control, enabled)
  File "/Users/swahal/.edm/envs/ets_tutorial/lib/python3.6/site-packages/traitsui/qt4/editor.py", line 95, in _e
nabled_changed_helper
    control.setEnabled(enabled)
RuntimeError: Internal C++ object (PySide2.QtWidgets.QPushButton) already deleted.
```

Note that `future.cancel()` doesn't immediately stop the future. It only moves it to a cancelling state.

Reply:
Not relevant anymore.